Landing Page

alonweissfeld opened the pull request:
This PR handles #5. 

It adds a basic Django app (`main_app`) that will serve the main application of the platform. With that, it adds a landing page that includes the basic structure of html templates and static css files. The css here is separated by files for the appropriate matching html templates.

This PR depends on #14. Therefore, **the files that are relevant for this PR review are mainly the files under** **` sellotape/sellotape_dj/main_app`**:
- `static` folder (css files)
- `templates` folder (html files)
-  `views.py`, `urls.py`
Other files within this directory are Django generated untouched files.

In addition, modifications (to review as well) were made to `settings.py` and `urls.py` under `sellotape/sellotape_dj/sellotape_dj` to include the new app in this Django project.

Commit:
Merged branch 11galpid:dev-userlogin

Commit:
Updated .gitignore to include MacOS .DS_Store.

Commit:
Initialized a Django app, which is the main sellotape project's app.

- Using Django statrapp command initialized the necessary files for the app
- Added the db.sqlite3 to gitignore

Commit:
Render initial index template.

Commit:
Added incusion of css static files.

Commit:
- Use 'extends' to extend and reuse html pages.
- Added basic platform description
- Nicer style

Commit:
Added sign-in button and custom css.

Commit:
Moved the landing page css to it's custom file.

Commit:
Updated README:

- Remove all related description of features that implement a live audio streaming within our platform,
as currently we will not support it.
- Updated some of the db tables columns.

Commit:
Removed auth_app from this branch.

In roder to reduce code changes and unrelated features auth_app (using django-social-app) was removed.
This includes the folder sellotape_dj/sellotape_dj/auth_app related url patterns (in sellotape_dj/sellotape_dj/urls.py) and removal of the app from the INSTALLED_APPS config in sellotape_dj/sellotape_dj/settings.py.

Commit:
Added ips to django's ALLOWED_HOSTS

Acceing the server from the host machine triggers ERR_CONNECTION_REFUSED since the VM binds to 127.0.0.1, which is loopback.
This commit makes sure that we can bind the server on the guest machine to 0.0.0.0 so that all interfaces can access it.
In addition, the local host ip was added since we've modified the default specification that includes it (and we still want to inculde it).

alonweissfeld commented on sellotape/sellotape_dj/sellotape_dj/settings.py:
```diff
@@ -0,0 +1,124 @@
+ALLOWED_HOSTS = [
+    '0.0.0.0',
+    '127.0.0.1',
+]
```
While running the server on the guest machine, accessing it through the host machine can often get an `ERR_CONNECTION_REFUSED`. Seems like this is caused by the server within the VM because it binds to `127.0.0.1`, which is loopback. We want to make sure that the server is bound to `0.0.0.0` so that all interfaces can access it.

https://stackoverflow.com/questions/18157353/connection-reset-when-port-forwarding-with-vagrant

boaz0 commented on sellotape/sellotape_dj/sellotape_dj/settings.py:
```diff
@@ -0,0 +1,124 @@
+# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'gtk7i_g+v%afq6dvplwms_-*r3j7uj-k6g4b&&iu#ys-e(ui)0'
```
Oh no! you just leaked an important information. Now any one knows your secret by searching on Github "SECRET".
Better set a new SECRET_KEY, remove this one and figure out a way to share it among you guys (maybe in a private chat on Slack and later in `Vault`)
Unless @lioramilbaum / @ifireball think it's OK.

ifireball replied:
Hold your horses, we're still on development environment here, so its ok... in production we would need to ensure the value comes from an environment variable that comes from a vault file or a k8s secret but we're not there yet.

@alonweissfeld you may want to create a future task for this though.

ifireball reviewed (changes requested):
I've merged PR #14 - please re-base this PR and lets see if it shrinks down to a manageable size, if not you may have to split it apart to smaller sets of changes.

Commit:
Removed whitespaces in README.

Commit:
Added new lines at end of files.

alonweissfeld commented:
@ifireball Pulled from master (it did shrink down). Toda

ifireball commented:
@alonweissfeld I still see 15 commits - did you rebase? Please try the following:

    # Assuming you configured the `upstream` remote to point to the main project repo
    # the following will hard-reset your local master to match it
    git checkout master
    git reset --hard upstream/master
    # Now to rebase your feture branch on the master
    git checkout landing-page
    git rebase master
    # in the output of the following command you should only see commits that contain
    # changes that are related to the new feature listed between the last commit of the
    # master branch and this branch
    git log --oneline --decorate
    # If all looks well you can force-push the new state to your fork, that will also cause the PR to
    # be updated
    git push --force origin landing-page

Commit:
Cleaned rebase/editor conflicts headers.

This branch started from merging a development branch (11galpid/dev-userlogin) that later was reduced, in order to simplify it's PR.
Therefore, while continuing development on this branch, it was removed as well at some point. This caused some turbulence in the existing
state of the common files between the branches. This commit is CLEANUP commit.

alonweissfeld commented:
@ifireball Except a few merges during development, all the commits here are part of this PR. I split the work to small commits for readability. 

I did try to rebase at first, but since this branch started from a branch that was later reduced - the rebased caused some conflicts which later were mistakenly saved in the editor. So, now, the rebase did clean some merge commits from before - all other 12 commits are part of the PR, and the last commit is for the cleanup (more details are in this last commit)

ifireball commented:
Ok, if you have a PR with 12 different commits and 17 changed files - you should know its too big already... do reviewers really need to read and understand 12 different steps to realize what you're trying to change and how?

I'm going to review and merge because it seems to be a big blocker for your team, but please improve in the future.

ifireball commented:
The right way to manage the introduction of auto-generated files to a project is to separate the addition of auto-generated files from manual changes. When you say a PR that everything in it is auto-generated, and clearly indicate it - then it can be reviewed and merged very quickly.

When you put the two together I have to ask myself which things are auto-generated, which did you write, and whihc are auto-generated things that you've later made manual changes to.

alonweissfeld commented:
@ifireball Got it. I will pay attention to this on the upcoming work. Thank you
